src/com/google/javascript/jscomp/js/es6/symbol.js

```diff
@@ -46,8 +46,30 @@ $jscomp.symbolCounter_ = 0;
  * @suppress {reportUnknownTypes}
  */
 $jscomp.Symbol = function(opt_description) {
-  return /** @type {symbol} */ (
-      $jscomp.SYMBOL_PREFIX + (opt_description || '') + ($jscomp.symbolCounter_++));
+  // List of ECMAScript 2015 "Well-known Symbols"
+  // See Table 1 in https://www.ecma-international.org/ecma-262/6.0/#sec-well-known-symbols
+  /** @const @dict @type {!Object<string, string>} */
+  var WELL_KNOWN_SYMBOLS = {
+    'hasInstance': '@@hasInstance',
```

Comment:
To throw a wrench in here, supporting Symbol.hasInstance and many others is a bit of a non-starter, since it would require rewriting distant files that have no ES6 syntax.

This is related to #2494, where we want to be able to support transpiling one file but not another.  So if we need to rewrite all calls to `instanceof` (or `for-in`, which would be required to transpile enumerability from ES5 to ES3) there's just no way to do it.